### PR TITLE
M3-3819 Longview user preferences

### PR DESCRIPTION
**Pull request by Jskobos**
## Description

Part I, only addressing the TimeRangeSelects used in Longview

- Return preferences data from the preferences container if no mStP is provided
- Add UserPreferences type to the SDK for easier reference
- Wrap TimeRangeSelect in the container, and default the select value to
the one stored in the longviewTimeRange preference
- Request + update preferences every time the time range is toggled

### CODE REVIEW

File: packages/manager/src/features/Longview/shared/TimeRangeSelect.tsx [hunk 1]
```diff
@@ -78,6 +84,15 @@ const TimeRangeSelect: React.FC<CombinedProps> = props => {
   const handleChange = (item: Item<Labels, Labels>) => {
     setTimeRange(item.value);
 
+    getUserPreferences()
```
Comment (Jskobos) [inline packages/manager/src/features/Longview/shared/TimeRangeSelect.tsx]: This is similar to what's in preferenceToggle, we should consider abstracting this up to the container.

Comment (Jskobos) [inline packages/manager/src/features/Longview/shared/TimeRangeSelect.tsx]: This is similar to what's in preferenceToggle, we should consider abstracting this up to the container.

File: packages/manager/src/features/Longview/shared/TimeRangeSelect.tsx [hunk 2]
```diff
@@ -40,7 +46,7 @@ const TimeRangeSelect: React.FC<CombinedProps> = props => {
     values when it comes time to make the request
   */
   const [selectedTimeRange, setTimeRange] = React.useState<Labels>(
-    defaultValue || 'Past 30 Minutes'
+    preferences?.longviewTimeRange || defaultValue || 'Past 30 Minutes'
```
Comment (johnwcallahan) [inline packages/manager/src/features/Longview/shared/TimeRangeSelect.tsx]: I think we should catalogue all possible Cloud user preference keys somewhere, either as a type or as constants or both.

File: packages/manager/src/store/preferences/preferences.actions.ts [hunk 3]
```diff
@@ -1,16 +1,31 @@
 import { APIError } from 'linode-js-sdk/lib/types';
 import { actionCreatorFactory } from 'typescript-fsa';
 
+import { SpacingChoice, ThemeChoice } from 'src/LinodeThemeWrapper';
+
 const actionCreator = actionCreatorFactory(`@@manager/preferences`);
 
+export interface UserPreferences {
+  longviewTimeRange?: string;
+  gst_banner_dismissed?: boolean;
```
Comment (Jskobos) [inline packages/manager/src/store/preferences/preferences.actions.ts]: Can't type in the SDK so the consequence of this is there are separate types for UserPreferences. I think it works out OK though

File: packages/manager/src/store/preferences/preferences.requests.ts [hunk 4]
```diff
@@ -40,11 +40,7 @@ export const updateUserPreferences: ThunkActionCreator<
 > = payload => dispatch => {
   const { started, done, failed } = handleUpdatePreferences;
 
-  dispatch(
-    started({
-      params: payload
-    })
-  );
+  dispatch(started(payload));
```
Comment (Jskobos) [inline packages/manager/src/store/preferences/preferences.requests.ts]: I don't entirely understand why this change is necessary to get the types to resolve, but the `params` object is unused in the reducer anyway.

File: packages/manager/src/store/preferences/preferences.requests.ts [hunk 5]
```diff
@@ -8,9 +8,9 @@ import {
   handleUpdatePreferences
 } from './preferences.actions';
 
-export const getUserPreferences: ThunkActionCreator<
-  Promise<Record<string, any>>
-> = () => dispatch => {
+export const getUserPreferences: ThunkActionCreator<Promise<
+  Record<string, any>
```
Comment (johnwcallahan) [inline packages/manager/src/store/preferences/preferences.requests.ts]: Related to your comment below, this should be typed as `UserPreferences` (as well as the other instances of `Record<string, any>` in this file).

Doing some digging, we should update the types in `preferences.container.ts` as well. I just realized there is already a `UserPreferences` type in the SDK which we should drop from the Cloud code in favor of your UserPreferences interface, I think.
